### PR TITLE
Add Graph-PIT Binary Cross-Entropy loss

### PR DESCRIPTION
**Pull request by thequilo**
Add the Graph-PIT Binary Cross-Entropy loss used in the paper "Utterance-by-utterance overlap-aware neural diarization with Graph-PIT" which is accepted at Interspeech 2022.

### CODE REVIEW

File: graph_pit/loss/optimized.py [hunk 1]
```diff
@@ -29,6 +47,11 @@ class OptimizedGraphPITLoss(GraphPITBase):
         [1] Speeding up Permutation Invariant Training
     """
     assignment_solver: Union[Callable, str] = 'optimal_dynamic_programming'
+    reduction: str = 'sum'
+
+    def __post_init__(self):
+        super().__post_init__()
+        assert self.reduction in ('sum', 'mean')
```
Comment (boeddeker) [inline graph_pit/loss/optimized.py:54]: Can you add `, self reduction` for a better error message?

File: graph_pit/loss/optimized.py [hunk 2]
```diff
@@ -82,6 +108,12 @@ class OptimizedGraphPITSourceAggregatedSDRLoss(OptimizedGraphPITLoss):
     """
     sdr_max: float = None
 
+    def __post_init__(self):
+        super().__post_init__()
+        assert self.reduction == 'sum', (
+            'Only sum reduction makes sense for SA-SDR'
+        )
```
Comment (boeddeker) [inline graph_pit/loss/optimized.py:115]: Can you add what the user requested in the error message?

File: graph_pit/loss/optimized.py (hunk 1 above)
Comment (thequilo) [reply graph_pit/loss/optimized.py]: Done

File: graph_pit/loss/optimized.py (hunk 2 above)
Comment (thequilo) [reply graph_pit/loss/optimized.py:115]: Done
